File: examples/HelloWorld.java
public class HelloWorld {
    public static void main(String[] args) {
        System.out.println("Hello, World from Java!");
        System.out.println("========================");
        System.out.println();

        // Variables and basic operations
        int a = 10, b = 5;
        System.out.println("Numbers: a = " + a + ", b = " + b);
        System.out.println("a + b = " + (a + b));
        System.out.println("a - b = " + (a - b));
        System.out.println("a * b = " + (a * b));
        System.out.println("a / b = " + (a / b));

        // String operations
        String greeting = "Welcome to Java programming!";
        System.out.println("\nString: " + greeting);
        System.out.println("String length: " + greeting.length());

        // Array demonstration
        int[] numbers = { 1, 2, 3, 4, 5 };
        System.out.print("\nArray contents: ");
        for (int num : numbers) {
            System.out.print(num + " ");
        }
        System.out.println();

        // Object creation
        String language = "Java";
        System.out.println("\nProgramming language: " + language.toUpperCase());
    }
}

File: examples/hello_world.cpp
#include <iostream>
#include <string>
#include <vector>

int main() {
    std::cout << "Hello, World from C++!" << std::endl;
    std::cout << "======================" << std::endl << std::endl;
    
    // Variables and basic operations
    int a = 10, b = 5;
    std::cout << "Numbers: a = " << a << ", b = " << b << std::endl;
    std::cout << "a + b = " << (a + b) << std::endl;
    std::cout << "a - b = " << (a - b) << std::endl;
    std::cout << "a * b = " << (a * b) << std::endl;
    std::cout << "a / b = " << (a / b) << std::endl;
    
    // String operations
    std::string greeting = "Welcome to C++ programming!";
    std::cout << "\nString: " << greeting << std::endl;
    std::cout << "String length: " << greeting.length() << std::endl;
    
    // Vector demonstration
    std::vector<int> numbers = {1, 2, 3, 4, 5};
    std::cout << "\nVector contents: ";
    for (int num : numbers) {
        std::cout << num << " ";
    }
    std::cout << std::endl;
    
    return 0;
}
